### PR TITLE
Enable unified UpdateCombinationCommand in product page v2

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | All the combination commands have been merged into a unified command It was previously behat tested, this is the final PR that integrates the new command into the forms (combination form, bulk edition, edition from list).
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| Related PRs       | ~
| How to test?      | Will have to make sure combination editionv2 works as usual. Only "simple fields" are impacted:<br>- Stocks<br>- Price impact<br>- References<br><br>The relationships (suppliers, images) are not impacted by this PR and don't need to be tested. The combination has three way of editions that all need to be tested:<br>- combination list<br>- combination form (accessed from list)<br>- combination bulk edition<br><br>This PR must NOT BE TESTED in multishop( this feature is still under development and is not in the scope of this PR)
| Possible impacts? | ~

### CODE REVIEW

**Comment by jolelievre** [comment]
Runnin tests https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/3572173334

File: src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml [hunk 1]
```diff
@@ -199,25 +199,14 @@ services:
       - '@prestashop.core.command_bus'
       - '@prestashop.core.admin.shop.repository'
 
-  prestashop.core.form.identifiable_object.data_handler.combination_form_data_handler:
-    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\CombinationFormDataHandler'
-    arguments:
-      - '@prestashop.core.command_bus'
-      - '@prestashop.core.form.command_builder.combination.combination_commands_builder'
+  PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\CombinationFormDataHandler:
+    autowire: true
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml:203]: probably first autowiring :tada:

**Review by zuk3975 — APPROVED** [review]

File: src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php [hunk 2]
```diff
@@ -83,6 +83,8 @@ public function enable(string $featureFlagName): void
         $featureFlag = $this->getByName($featureFlagName);
         if (null !== $featureFlag) {
             $featureFlag->enable();
```
Comment (lartist) [inline src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php:85]: Why are we not using a command to enable feature? The domain logic is in a repository, it' strongly coupled ! I know it's not on the PR scope. I'am just wondering why?

File: src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php [hunk 3]
```diff
@@ -96,6 +98,8 @@ public function disable(string $featureFlagName): void
         $featureFlag = $this->getByName($featureFlagName);
         if (null !== $featureFlag) {
             $featureFlag->disable();
```
Comment (lartist) [inline src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php:100]: Same here

**Review by lartist — APPROVED** [review]

File: src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php (hunk 2 above)
Comment (zuk3975) [reply src/PrestaShopBundle/Entity/Repository/FeatureFlagRepository.php:85]: Yep indeed, Im pretty sure we could (and should?) have a cqrs command for this :thinking:

**Review by matthieu-rolland — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @jolelievre ,

Tested the following on `develop` with PPV2 enabled:

In BO, checked combination edition on those fields only:
- Stocks
- Price impact//Retail Price
- References//Specific References
Screenshot 2022-11-30 at 15 46 05

On different mediums:
- Combination list
- Combination form (accessed from list)
- Combination bulk edition

Each time checked edition on 1 field, multiple fields and all fields when possible.

Modifications are well taken into account in combination list and combination form. ✔️
Checked with error in used fields, error is well displayed. ✔️
Check multiple combination and modified directly field in list. ✔️
Save & Publish can't be clicked when Edition is not saved before. ✔️
Checked Edition mode buttons : Cancel, Reset and Save list. ✔️
Edition combination right panel is well updated after each modification. ✔️

I have just checked the Advanced Parameters > Logs page, I have some error displayed while testing PPV2.
Could you create another PR for that ? I think it is out-of-scope.

Screenshot 2022-11-30 at 16 16 42

_The parameter table in function existsInDatabase (Line 54) is deprecated and will be removed in the next major version._

NOT TESTED:
- Suppliers
- Images
- Multishop

It is QA ✅ for me !
Thanks!

**Comment by zuk3975** [comment]
:rocket:

**Pull request merged**